src/main/groovy/com/github/jk1/license/reader/ProjectReader.groovy

```diff
@@ -20,36 +18,44 @@ class ProjectReader {
     ProjectData read(Project project) {
         ProjectData data = new ProjectData()
         data.project = project
-        // Get the configurations matching the name: that's our base set
-        final Set<Configuration> toReport = new HashSet<Configuration>(project
-                .getConfigurations().matching(new Spec<Configuration>() {
-            @Override
-             boolean isSatisfiedBy(Configuration configuration) {
-                for (String configurationName : project.licenseReport.configurations) {
-                    if (configuration.getName().equalsIgnoreCase(configurationName)) {
-                        if (configuration.hasProperty("canBeResolved") && !configuration.canBeResolved) {
-                            throw new GradleException("Project Reader: " +
-                                    "The specified configuration \"${configuration.name}\" can't be resolved. " +
-                                    "Try specifying a more specific configuration by adding flavor(s) and/or build type.")
-                        }
-                        return true
-                    }
-                }
-                return false
-            }
-        }))
-
-        // Now, keep adding extensions until we don't change the set size
-        for (int previousRoundSize = 0; toReport.size() != previousRoundSize; previousRoundSize = toReport.size()) {
-            for (Configuration configuration : new ArrayList<Configuration>(toReport)) {
-                toReport.addAll(configuration.getExtendsFrom())
-            }
-        }
-        LOGGER.info("Configurations: " + toReport.join(','))
-        for (Configuration configuration : toReport) {
+
+        final Set<Configuration> configurationsToScan = findConfigurationsToScan(project)
+
+        configurationsToScan.addAll(getAllExtendedConfigurations(configurationsToScan))
+
+        LOGGER.info("Configurations: " + configurationsToScan.join(','))
+        for (Configuration configuration : configurationsToScan) {
             LOGGER.info("Reading configuration: " + configuration)
             data.configurations.add(configurationReader.read(project, configuration))
         }
         return data
     }
+
+    private static Set<Configuration> getAllExtendedConfigurations(Set<Configuration> configurationsToScan) {
+        configurationsToScan.collect { it.extendsFrom }.flatten()
+    }
+
+    private static Set<Configuration> findConfigurationsToScan(Project project) {
+        project.configurations.findAll { config ->
+            isConfigured(config, project) && isResolvable(config)
```

Comment:
I would distinguish this cases:
* Configuration is defined in the gradle script (or non which defaults to `runtime`) & configs are resolvable -> us those configs
* Configuration is defined in the gradle script (or non which defaults to `runtime`) & one config is not resolvable -> throw an Exception that the defined configuration can't be resolved.
* Configurations are set to empty (so use all configurations by default) & configs are resolvable -> use those configs
* Configurations are set to empty (so use all configurations by default) & one config is not resolvable  -> ignore that config and use the rest

It is for sure confusing, if a config is defined (e.g. `runtimeOnly`) and the report then is just empty. A exception should be thrown instead which states that that config is not resolvable.

Reply:
Alright, sounds reasonable.